src/main/java/com/openlattice/shuttle/dates/JavaDateTimeHelper.java

```diff
@@ -1,146 +1,183 @@
-    private static final Logger logger = LoggerFactory.getLogger(JavaDateTimeHelper.class);
+    private static final Logger logger = LoggerFactory.getLogger( JavaDateTimeHelper.class );
 
-    private final TimeZone      tz;
-    private final String[]      datePatterns;
+    private       ZoneId   zoneId;
+    private       boolean  shouldAddTimezone;
+    private final String[] datePatterns;
 
-    public JavaDateTimeHelper(TimeZone tz, String... datePatterns) {
-        this.tz = tz;
+    public JavaDateTimeHelper( Optional<TimeZone> tz, String... datePatterns ) {
         this.datePatterns = datePatterns;
+        this.zoneId = tz.orElse( Constants.DEFAULT_TIMEZONE ).toZoneId();
+        this.shouldAddTimezone = tz.isPresent();
     }
 
-    private boolean shouldIgnoreValue(String date) {
-        return date == null || StringUtils.isBlank(date) || date.equals("NULL");
+    /**
+     * Parse datetime into OffsetDateTime:
+     * - parse into OffsetDateTime as ISO + check whether timezones match
+     * - parse into OffsetDateTime with provided patterns + perform 2-digit-year-fix + check whether timezones match
+     * - parse into LocalDateTime as ISO + add timezone
+     * - parse into LocalDateTime with provided patterns + perform 2-digit-year-fix + add timezone
+     *
+     * @param date - String to parse
+     */
+    public OffsetDateTime parseDateTime( String date ) {
+
+        if ( StringUtils.isBlank( date ) || date.equals( "NULL" ) )
+            return null;
+
+        // Try parsing into a OffsetDateTime
+        OffsetDateTime odt = parseFromOffsetDateTime( date );
+        if ( shouldAddTimezone )
+            TimeZones.checkTimezonesMatch( odt, zoneId );
+        if ( odt != null )
+            return odt;
+
+        // Try parsing into OffsetDateTime with patterns
+        OffsetDateTime odt_p = parseFromPatterns(
+                date,
+                ( toParse, formatter ) -> OffsetDateTime.parse( toParse, formatter ),
+                ( local_odt, datePattern ) -> DecadeChangeHelper
+                        .fixTwoYearPatternOffsetDateTime( local_odt, datePattern ) );
+        if ( shouldAddTimezone )
+            TimeZones.checkTimezonesMatch( odt, zoneId );
+        if ( odt_p != null )
+            return odt_p;
+
+        // Try parsing into a LocalDateTime
+        LocalDateTime ldt = parseFromLocalDateTime( date );
+        if ( ldt != null )
+            return ldt.atZone( zoneId ).toOffsetDateTime();
+
+        // Try parsing into a LocalDateTime with patterns
+        LocalDateTime ldt_p = parseFromPatterns(
+                date,
+                ( toParse, formatter ) -> LocalDateTime.parse( toParse, formatter ),
+                ( local_odt, datePattern ) -> DecadeChangeHelper.fixTwoYearPatternLocalDate( local_odt, datePattern ) );
+        if ( ldt_p != null )
+            return ldt_p.atZone( zoneId ).toOffsetDateTime();
+
+        logger.error( "Could not parse Date Time " + date );
```

Comment:
Same minor comment as for other logs with additional arguments, you can log it like this:
`logger.error( "Could not parse Date Time {}", date );`